src/providers/moduleProvider.ts

```diff
@@ -0,0 +1,161 @@
+    }
+  }
+
+  iconPath = this.getIcon(this.type);
+
+  getIcon(type: string) {
+    const icon = this.terraformIcon();
+    switch (type) {
+      case 'tfregistry':
+        return icon;
+      case 'local':
+        return LOCALMODULE;
+      case 'github':
+        return GITHUBMODULE;
+      default:
+        return TFREGISTRY;
+    }
+  }
+
+  private terraformIcon() {
+    // need current extension path to find icon svg
+    // could possibly make this a custom icon
+    const myExtDir = vscode.extensions.getExtension('hashicorp.terraform').extensionPath;
+    const svg = vscode.Uri.file(path.join(myExtDir, 'assets', 'icons', 'terraform.svg'));
+    const icon = {
+      light: svg,
+      dark: svg,
+    };
+    return icon;
+  }
+}
+
+export class TerraformModuleProvider implements vscode.TreeDataProvider<TerraformModule> {
+  private _onDidChangeTreeData: vscode.EventEmitter<TerraformModule | undefined | null | void> =
+    new vscode.EventEmitter<TerraformModule | undefined | null | void>();
+  readonly onDidChangeTreeData: vscode.Event<TerraformModule | undefined | null | void> =
+    this._onDidChangeTreeData.event;
+
+  constructor(ctx: vscode.ExtensionContext, public handler: ClientHandler) {
+    ctx.subscriptions.push(
+      vscode.commands.registerCommand('terraform.modules.refreshList', () => this.refresh()),
+      vscode.commands.registerCommand('terraform.modules.documentation', (module: TerraformModule) => {
+        vscode.env.openExternal(vscode.Uri.parse(module.doclink));
+      }),
+      vscode.window.onDidChangeActiveTextEditor(async (event: vscode.TextEditor | undefined) => {
+        if (event && vscode.workspace.workspaceFolders[0]) {
+          this.refresh();
+        }
+      }),
+    );
+  }
+
+  refresh(): void {
+    this._onDidChangeTreeData.fire();
+  }
+
+  getTreeItem(element: TerraformModule): TerraformModule | Thenable<TerraformModule> {
+    return element;
+  }
+
+  getChildren(element?: TerraformModule): vscode.ProviderResult<TerraformModule[]> {
+    if (element) {
+      return Promise.resolve(element.children);
+    } else {
+      const m = this.getModules();
+      return Promise.resolve(m);
+    }
+  }
+
+  getCollapseState(type: string): vscode.TreeItemCollapsibleState {
+    switch (type) {
+      case 'tfregistry':
+        return vscode.TreeItemCollapsibleState.Collapsed;
+      case 'local':
+        return vscode.TreeItemCollapsibleState.None;
+      case 'github':
+        return vscode.TreeItemCollapsibleState.None;
+      default:
+        return vscode.TreeItemCollapsibleState.None;
+    }
+  }
+
+  async getModules(): Promise<TerraformModule[]> {
+    const activeEditor = vscode.window.activeTextEditor;
+    if (activeEditor === undefined) {
+      return Promise.resolve([]);
+    }
+
+    const document = activeEditor.document;
+    if (document === undefined) {
+      return Promise.resolve([]);
+    }
+
+    const editor = document.uri;
```

Comment:
similar to above you'll probably want to leverage `getActiveTextEditor` from https://github.com/hashicorp/vscode-terraform/pull/771 here